include/grpc++/impl/codegen/client_context.h

```diff
@@ -307,6 +307,12 @@ class ClientContext {
   };
   static void SetGlobalCallbacks(GlobalCallbacks* callbacks);
 
+  // Should be used for framework-level extensions only.
+  // Applications never need to call this method.
+  grpc_call* c_call() const {
```

Comment:
In C we can easily extend the call API by exposing such a method:
```
grpc_my_extension* grpc_call_get_my_extension(grpc_call* call);
```

In C++, the `Server/ClientContext` is a class which is much harder to extend. @ctiller and I discussed this already. We *could* have added a macro in the class definition but that felt really yucky. This alternative solution will allow us to do something like:
```
grpc::MyExtension* GetMyExtensionFromServerContext(grpc::ServerContext* ctx);
```
which seems better overall.